applications/GeoMechanicsApplication/custom_strategies/schemes/geomechanics_time_integration_scheme.hpp

```diff
@@ -0,0 +1,360 @@
+    using LocalSystemMatrixType = typename BaseType::LocalSystemMatrixType;
+
+    void GetDofList(const Element& rElement,
+                    Element::DofsVectorType& rDofList,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (IsActive(rElement))
+            rElement.GetDofList(rDofList, rCurrentProcessInfo);
+    }
+
+    void GetDofList(const Condition& rCondition,
+                    Condition::DofsVectorType& rDofList,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (IsActive(rCondition))
+            rCondition.GetDofList(rDofList, rCurrentProcessInfo);
+    }
+
+    void EquationId(const Element& rElement,
+                    Element::EquationIdVectorType& rEquationId,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (IsActive(rElement))
+            rElement.EquationIdVector(rEquationId, rCurrentProcessInfo);
+    }
+
+    void EquationId(const Condition& rCondition,
+                    Condition::EquationIdVectorType& rEquationId,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (IsActive(rCondition))
+            rCondition.EquationIdVector(rEquationId, rCurrentProcessInfo);
+    }
+
+    void Initialize(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        SetTimeFactors(rModelPart);
+
+        Scheme<TSparseSpace, TDenseSpace>::mSchemeIsInitialized = true;
+
+        KRATOS_CATCH("")
+    }
+
+    void InitializeSolutionStep(ModelPart& rModelPart,
+                                TSystemMatrixType& A,
+                                TSystemVectorType& Dx,
+                                TSystemVectorType& b) override
+    {
+        KRATOS_TRY
+
+        SetTimeFactors(rModelPart);
+
+        const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+
+        block_for_each(rModelPart.Elements(), [&r_current_process_info, this](Element& rElement) {
+            if (IsActive(rElement))
+                rElement.InitializeSolutionStep(r_current_process_info);
+        });
+
+        block_for_each(rModelPart.Conditions(), [&r_current_process_info,
+                                                 this](Condition& rCondition) {
+            if (IsActive(rCondition))
+                rCondition.InitializeSolutionStep(r_current_process_info);
+        });
+
+        KRATOS_CATCH("")
+    }
+
+    void Predict(ModelPart& rModelPart,
+                 DofsArrayType& rDofSet,
+                 TSystemMatrixType& A,
+                 TSystemVectorType& Dx,
+                 TSystemVectorType& b) override
+    {
+        this->UpdateVariablesDerivatives(rModelPart);
+    }
+
+    void InitializeNonLinIteration(ModelPart& rModelPart,
+                                   TSystemMatrixType& A,
+                                   TSystemVectorType& Dx,
+                                   TSystemVectorType& b) override
```

Comment:
And the same thing here:
```suggestion
                                   TSystemMatrixType&,
                                   TSystemVectorType&,
                                   TSystemVectorType&) override
```

Reply:
Done